app/assets/javascripts/markdown_editor.js

```diff
@@ -0,0 +1,98 @@
+/**
+ * ToastUi editor initializer
+ *
+ * This script transforms form textareas created with 
+ * "MarkdownFormBuilder" into ToastUi markdown editors.
+ *
+ */
+
+const initializeMarkdownEditors = () => {
+  const editors = document.querySelectorAll(
+    '[data-behavior="markdown-editor-wrapper"]'
+  );
+
+  editors.forEach((wrapper) => {
+    const editor = wrapper.querySelector(
+      '[data-behavior="markdown-editor-widget"]'
+    );
+    const formInput = wrapper.querySelector(
+      '[data-behavior="markdown-form-input"]'
+    );
+
+    if (!editor || !formInput) return;
+
+    const toastEditor = new ToastUi({
+      el: editor,
+      theme: window.getCurrentTheme(),
+      initialValue: formInput.value,
+      placeholder: formInput.placeholder,
+      extendedAutolinks: true,
+      height: "400px",
+      autofocus: false,
+      usageStatistics: false,
+      language: I18n.locale,
+      toolbarItems: [
+        ["heading", "bold", "italic"],
+        ["link", "quote", "code"],
+        ["image"],
+        ["ul", "ol"],
+      ],
+      initialEditType: "markdown",
+      events: {
+        change: () => {
+          // Keep real form <textarea> in sync
+          const content = toastEditor.getMarkdown();
+          formInput.value = content;
+        },
+      },
+    });
+
+    // Prevent user from drag'n'dropping images in the editor
+    toastEditor.removeHook("addImageBlobHook");
+
+    // Delegate focus from form input to toast ui editor
+    formInput.addEventListener("focus", () => {
+      toastEditor.focus();
+    });
+  });
+};
+
+const disableImageUpload = () => {
+  // Reference:https://github.com/nhn/tui.editor/issues/1204#issuecomment-1068364431
+  const observer = new MutationObserver(() => {
+    if (
+      document.querySelector(
+        ".toastui-editor-popup.toastui-editor-popup-add-image"
+      )
+    ) {
+      document.querySelector('[aria-label="URL"]').click();
+      document.querySelector('[aria-label="File"]').style.display = "none";
+    }
+  });
+
+  const target = document.querySelector(".toastui-editor-popup ");
+  observer.observe(target, { attributes: true, attributeFilter: ["style"] });
+};
+
+const setMakrdownEditorTheme = (theme) => {
```

Comment:
Nice, many thanks for adding theme support -- that's great! ☀️🌙

```suggestion
const setMarkdownEditorTheme = (theme) => {
```